src/Microsoft.AspNet.Mvc.Abstractions/ModelBinding/ModelStateDictionary.cs

```diff
@@ -869,6 +880,55 @@ public void Reset()
                 _enumerator = default(Dictionary<string, ModelStateEntry>.Enumerator);
                 Current = default(KeyValuePair<string, ModelStateEntry>);
             }
+
+        }
+
+        /// <summary>
+        /// Enumerates the elements of a <see cref="ModelStateDictionary"/>.
+        /// </summary>
+        public struct Enumerator : IEnumerator<KeyValuePair<string, ModelStateEntry>>
+        {
+            private Dictionary<string, ModelStateEntry>.Enumerator _enumerator;
+
+            internal Enumerator(Dictionary<string, ModelStateEntry>.Enumerator enumerator)
+            {
+                _enumerator = enumerator;
+            }
+
+            /// <inheritdoc />
+            public KeyValuePair<string, ModelStateEntry> Current
+            {
+                get
+                {
+                    return _enumerator.Current;
+                }
+            }
+
+            /// <inheritdoc />
+            object IEnumerator.Current
+            {
+                get
+                {
+                    return _enumerator.Current;
+                }
+            }
+
+            /// <inheritdoc />
+            public void Dispose()
+            {
+                _enumerator.Dispose();
+            }
+
+            /// <inheritdoc />
+            public bool MoveNext()
+            {
+                return _enumerator.MoveNext();
+            }
+
+            /// <inheritdoc />
+            public void Reset()
+            {
+            }
```

Comment:
There is no Reset() on the underlying enumerator. I can change it to throw an exception()